### DISCUSSION TITLE
Impact of the learning rate

### DISCUSSION THREAD

**Discussion by VForiel** [Leirof/M2-Unveiling-3D-structure-of-pre-stellar-cores#47] [Results]
I finally found the reason of the spikes that appear in #44 : the learning rate was too high.

I made several trainings with different learning rate in order to see how it impact the convergeance.

It result that 1e-3 is too high in the considered problems, leading to spikes in the convergeance process.

1e-5 is very stable, but it takes more time to reach the same accuracy

1e-4 seems to be a good compromise between stability and accuracy

Here is the tests of these 3 learning rates with an adam optimizing function
![image](https://github.com/Leirof/M2-Unveiling-3D-structure-of-pre-stellar-cores/assets/12165342/dcd65283-01e6-419e-acb0-a5d944f5f995)

And here is the same tests with an RMSprop optimizing function
![image](https://github.com/Leirof/M2-Unveiling-3D-structure-of-pre-stellar-cores/assets/12165342/1ab58fde-0664-4024-8d6e-e6eda49f55e0)

If we compare the two optimizing function with the learning rate at 1e-4, we see that the adam optimizer is naturally more stable and is able to converge to a better accuracy if we let it train during an sufficient time (I extrapolate a bit because these traning take time and I'm now limited in time as I approach to the end of the internship, but such extrapolation is not that risky due to the fact that there is almost never rapide change of behavior in the convergeance process... except spikes)

![image](https://github.com/Leirof/M2-Unveiling-3D-structure-of-pre-stellar-cores/assets/12165342/7a0da43f-c29c-4c8f-ab75-2996ffbf35b0)
